### PR TITLE
fix: inject annotations for service

### PR DESCRIPTION
**Pull request by Gala-R**

### CODE REVIEW

File: apis/apps/v2alpha1/util.go [hunk 1]
```diff
@@ -67,3 +67,12 @@ func MergeServicePorts(ports1, ports2 []corev1.ServicePort) []corev1.ServicePort
 
 	return result
 }
+
+func MergeMap(dst, src map[string]string) map[string]string {
```
Comment (Rory-Z) [inline apis/apps/v2alpha1/util.go:71]: move it to controllers

File: controllers/apps/v2alpha1/generated.go [hunk 2]
```diff
@@ -158,6 +169,17 @@ func generateListenerService(instance *appsv2alpha1.EMQX, listenerPorts []corev1
...
+	instanceAnnotations := instance.Annotations
+	if instanceAnnotations == nil {
+		instanceAnnotations = make(map[string]string)
+	}
+	delete(instanceAnnotations, "kubectl.kubernetes.io/last-applied-configuration")
+
+	serviceAnnotions := instance.Spec.ListenersServiceTemplate.Annotations
+	if serviceAnnotions == nil {
+		serviceAnnotions = make(map[string]string)
+	}
```
Comment (Rory-Z) [inline controllers/apps/v2alpha1/generated.go:181]: `make(map[string]string)` can running in `mergeMap` function

File: apis/apps/v2alpha1/util.go (hunk 1 above)
Comment (Gala-R) [reply apis/apps/v2alpha1/util.go:71]: > move it to controllers

have done it

File: controllers/apps/v2alpha1/generated.go (hunk 2 above)
Comment (Gala-R) [reply controllers/apps/v2alpha1/generated.go]: > `make(map[string]string)` can running in `mergeMap` function

have done it

File: controllers/apps/v2alpha1/generated.go [hunk 3]
```diff
@@ -136,7 +142,7 @@ func generateDashboardService(instance *appsv2alpha1.EMQX) *corev1.Service {
 			Namespace:   instance.Namespace,
 			Name:        instance.Spec.DashboardServiceTemplate.Name,
 			Labels:      instance.Spec.DashboardServiceTemplate.Labels,
-			Annotations: instance.Spec.DashboardServiceTemplate.Annotations,
+			Annotations: mergeMap(instance.Spec.DashboardServiceTemplate.Annotations, instanceAnnotations),
```
Comment (Rory-Z) [inline controllers/apps/v2alpha1/generated.go:145]: Sorry, my bad, I think this merge should in default web hook

File: controllers/apps/v2alpha1/generated.go [hunk 4]
```diff
@@ -167,7 +179,7 @@ func generateListenerService(instance *appsv2alpha1.EMQX, listenerPorts []corev1
 			Namespace:   instance.Namespace,
 			Name:        instance.Spec.ListenersServiceTemplate.Name,
 			Labels:      instance.Spec.ListenersServiceTemplate.Labels,
-			Annotations: instance.Spec.ListenersServiceTemplate.Annotations,
+			Annotations: mergeMap(instance.Spec.ListenersServiceTemplate.Annotations, annotations),
```
Comment (Rory-Z) [inline controllers/apps/v2alpha1/generated.go:182]: I think this merge should in default web hook

Comment (Gala-R) [reply controllers/apps/v2alpha1/generated.go:182]: > I think this merge should in default web hook

ok, I agree with you

File: controllers/apps/v2alpha1/generated.go (hunk 3 above)
Comment (Gala-R) [reply controllers/apps/v2alpha1/generated.go]: > Sorry, my bad, I think this merge should in default web hook

done

**Comment by Rory-Z** [comment]
the v1beta3 need this fix ?

**Comment by Gala-R** [comment]
> the v1beta3 need this fix ?

v1beta3 don't hava this problem

File: apis/apps/v2alpha1/util_test.go [hunk 5]
```diff
@@ -134,3 +134,25 @@ func TestMergeServicePorts(t *testing.T) {
 
 	assert.Equal(t, expect, MergeServicePorts(ports1, ports2))
 }
+
+func TestMap(t *testing.T) {
```
Comment (Rory-Z) [inline apis/apps/v2alpha1/util_test.go:138]: func TestMergeMap

**Review by Rory-Z — APPROVED** [review]

File: apis/apps/v2alpha1/util_test.go (hunk 5 above)
Comment (Gala-R) [reply apis/apps/v2alpha1/util_test.go]: > func TestMergeMap

Done

**Pull request merged**